File: 3861-minimum-capacity-box/3861-minimum-capacity-box.java
class Solution {
    public int minimumIndex(int[] capacity, int itemSize) {
        int n = capacity.length;
        PriorityQueue<int[]> pq = new PriorityQueue<>((a, b) -> {
            if (a[0] == b[0])
                return a[1] - b[1];

            return a[0] - b[0];
        }); // {cap, idx};
        for (int i = 0; i < n; i++) {
            pq.offer(new int[] { capacity[i], i });
        }

        int ans = -1;
        while (!pq.isEmpty()) {
            int c = pq.peek()[0];
            int id = pq.peek()[1];
            pq.poll();
            if (c >= itemSize) {
                ans = id;
                break;
            }
        }
        return ans;
    }
}

File: 3861-minimum-capacity-box/3861-minimum-capacity-box.py
class Solution:
    def minimumIndex(self, capacity: list[int], itemSize: int) -> int:
        n = len(capacity)
        h = []

        for i in range(n):
            h.append([capacity[i], i])
        
        h.sort(key = lambda x : x[0])
        ans = -1

        for i in range(n):
            [c, idx] = h[i]
            if c >= itemSize:
                ans = idx
                break
        
        return ans